main.go

```diff
@@ -43,43 +43,38 @@ func (*ExprHelp) Synopsis() string { return "Help about package expressions" }
 func (*ExprHelp) Usage() string {
 	return `Package expressions allow to specify calculations with dependencies.
 
-The examples use X, Y and Z as the package names, however they can be replaced
-by any expression.
-
-Examples using Q can only be used on packages and not expressions.
+The examples use X, Y and Z as placeholders for packages, packages paths or
+package expressions.
 
 # Basic operations
 	There are a few basic oprations specified for manipulating sets of packages.
 
 	X Y Z;
 	X + Y + Z;  add(X, Y, Z);  or(X, Y, Z)
-		packages that are used by X, Y or Z
+		all packages that match X, Y and Z
 
 	X - Y - Z;  subtract(X, Y, Z);  exclude(X, Y, Z)
 		packages that are used by X and not used by Y and Z
 
 	shared(X, Y, Z);  intersect(X, Y, Z)
-		packages that are used by all X, Y and Z
+		packages that exist in all of X, Y and Z
 
 	xor(X, Y);
 		packages that are different between X and Y
 
 # Selectors
-	Selectors allow selecting parts of the dependency tree.
+	Selectors allow selecting parts of the dependency tree
 
-	Q:root
-		packages that are included by package Q without dependencies
-	Q:noroot
-		selects excluding roots, shorthand for (Q - Q:root)
+	X:all
+		select X and all of its dependencies, shorthand for (X + X:deps)
+	X:deps
+		select dependenices of X
```

Comment:
```suggestion
dependencies
```

This should make it clear if it selects all dependencies recursively, or just the top level (direct) dependencies.

I can see arguments in both directions:

- with the former, you can get the latter with `X:deps - X:deps:deps`
- with the latter, you can get the former with `X:deps:all`

Since the non-recursive behaviour is now the default for package lists, maybe it might make sense to have non-recursive behaviour for `:deps` too.

Reply:
I think we can just add another selector. That way you can use both easily. Although, that doesn't solve the naming question.

Maybe `:imports` for the direct dependencies? I cannot think a nice shorter name.

Another name could be: `:deps` for direct dependencies and `:down` (for all downstream packages).

Reply:
I'm thinking that `:deps:all` is easy to type, logically clear, and only one more character to type than `:imports`.

Also, my intuition here is that it's nicely consistent if you _only_ get all recursive dependencies if you use `:all`. Nice and easy to explain. This also potentially makes things faster, as it's quicker to evaluate an expression if it doesn't need to recurse.

And no new name to invent.

Reply:
Fair enough.

Reply:
Although, I'm wondering whether it should be called `:imps` anyways. Maybe it's clearer than "dependencies", which is used as both direct and indirect dependencies. However "imports" is clearer in that regard.

Reply:
I think I agree about the naming, but how about just calling it `:imports`? That's super clear ("the set of directly imported packages"), and whole words are easy to type, and really easy to explain. You could add a `:imps` alias too, but I'd only do that if the length of the `imports` word was really turning out to be an issue.

Reply:
I mean `deps` is a shorthand as well :D. I think I'll add both... I know that I would rather type `:imps` than `:imports`.